prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/hx_insight/PredictionCollector.java

```diff
@@ -97,64 +86,65 @@ public void configure()
-                .choice().when(bodyNotEmpty())
+                .choice().when(statusCodeNot204())
+                    .log(DEBUG, log, "Processing prediction batches page ${headers.%s} started".formatted(BATCHES_PAGE_NO_HEADER))
                     .unmarshal(predictionsBatchDataFormat)
                     .split(body())
-                        .setHeader(BATCH_ID_HEADER, simple("${body.id}"))
-                        // notify HxI -> started batch processing
-                        .setHeader(PREDICTIONS_PAGE_NO_HEADER, constant(1))
-                        .loopDoWhile(bodyNotEmpty())
-                            .log(DEBUG, log, "Calling URL: " + predictionsUrl)
-                            .toD(predictionsUrl)
-                            .log(DEBUG, log, "Sending predictions to internal buffer: ${body}")
-                            .choice().when(bodyNotEmpty())
-                                .unmarshal(predictionsDataFormat)
-                                .split(body())
-                                    .marshal(predictionDataFormat)
-                                    .log(TRACE, log, "Sending prediction to internal buffer: ${body}")
-                                    .to(insightPredictionsProperties.bufferEndpoint())
-                                .end()
-                            .end()
-                            .setHeader(PREDICTIONS_PAGE_NO_HEADER).spel("#{request.headers['%s'] + 1}".formatted(PREDICTIONS_PAGE_NO_HEADER))
-                        .end()
-                        // notify HxI -> finished batch processing
+                        .to(BATCH_PROCESSOR_ENDPOINT)
+                    .end()
+                    .setHeader(BATCHES_PAGE_NO_HEADER, spel("#{request.headers['%s'] + 1}".formatted(BATCHES_PAGE_NO_HEADER)))
+                    .toD(batchesUrl)
+                .end()
+            .end()
+            .log(DEBUG, log, "Finished processing predictions");
+
+        from(BATCH_PROCESSOR_ENDPOINT)
+            .routeId(BATCH_PROCESSOR_ROUTE_ID)
+            .log(DEBUG, log, "Processing prediction batch ${body.id} started")
+            .setHeader(BATCH_ID_HEADER, simple("${body.id}"))
+            .setHeader(PREDICTIONS_PAGE_NO_HEADER, constant(1))
+            .loopDoWhile(statusCodeNot204())
+                .toD(predictionsUrl)
+                .choice().when(statusCodeNot204())
+                    .log(TRACE, log, "Processing page ${headers.%s} of predictions in batch ${headers.%s}, ${body}, ${header.CamelHttpResponseCode}".formatted(PREDICTIONS_PAGE_NO_HEADER, BATCH_ID_HEADER))
+                    .unmarshal(predictionsDataFormat)
+                    .split(body())
+                        .marshal(predictionDataFormat)
+                        .to(insightPredictionsProperties.bufferEndpoint())
                     .end()
+                    .setBody(simple("{\"status\": \"COMPLETE\", \"currentPage\": ${headers.%s}}".formatted(PREDICTIONS_PAGE_NO_HEADER)))
```

Comment:
I think (but am not sure) that the status shouldn't be `COMPLETE` until the last page has been processed. Let's check on the call in 15 mins.

Reply:
The HxI team are going to get back to us about the set of valid values for this field. We're expecting it to do something like:
```
APPROVED -> IN PROGRESS -> COMPLETE
APPROVED: Ready to start
IN PROGRESS: Not all pages in AMQ
COMPLETE: All pages in AMQ
```

Reply:
Here's the message from Ryan: https://alfresco.slack.com/archives/C057YBK8H8D/p1716904597681529

Reply:
yep, you're right, I also confirmed with Ryan, fixed